web/docs/project/env-vars.md

```diff
@@ -92,40 +93,44 @@ These files should not be committed to version control, and they are already ign
+While in development, we had the option of using `.env` files which made it easy to define and manage env vars. However, in production, when our code is built and deployed, we need to provide env vars differently.
 
 ![Env vars usage in development and production](/img/env/prod_dev_fade_2.svg)
 
 ### Client Env Vars
 
-Client env vars are embedded into the client code during the build and shipping process, making them public and readable by anyone. Therefore, you should **never store secrets in them** (such as secret API keys).
+Client env vars are embedded into the client code during the build process, making them public and readable by anyone. Therefore, you should **never store secrets in them** (such as secret API keys).
 
-You should provide them to the build command, for example:
+You should provide them to the build command that turns client code into static files:
 ```shell
-REACT_APP_SOME_VAR_NAME=somevalue npm run build
+REACT_APP_SOME_VAR_NAME=somevalue REACT_APP_SOME_OTHER_VAR_NAME=someothervalue npm run build
 ```
 
+Notice that the exact moment when you should provide these env vars is during the `npm run build` command that you execute from `.wasp/build/web-app/` after you have ran `wasp build`. Check [deployment docs](../advanced/deployment/manually#3-deploying-the-web-client-frontend) for more details.
+
+Also, notice that you can't and shouldn't provide env vars to the client code by setting them on the hosting provider where you deployed them (unlike server env vars, where this is how we do it). Your client code will ignore those, as at that point client code is just static files.
+
 :::info How it works
-What happens behind the scenes is that Wasp will replace all occurrences of `import.meta.env.REACT_APP_SOME_VAR_NAME` with the value you provided. This is done during the build process, so the value is embedded into the client code.
+What happens behind the scenes is that Wasp will replace all occurrences of `import.meta.env.REACT_APP_SOME_VAR_NAME` in your client code with the env var value you provided. This is done during the build process, so the value is embedded into the static files produced from the client code.
 
 Read more about it in Vite's [docs](https://vitejs.dev/guide/env-and-mode.html#production-replacement).
 :::
 
 ### Server Env Vars
 
-The way you provide env vars to your Wasp project in production depends on where you deploy it. For example, if you deploy your project to [Fly](https://fly.io), you can define them using the `flyctl` CLI tool:
+To provide env vars to your server code in production you will need to have them defined and available on the server where your server code is running.
+
+Setting this up will highly depend on where you are deploying your Wasp project, but in general it comes down to defining the env vars via mechanisms that your hosting provider provides.
```

Comment:
I recommend splitting into two sentences.

Reply:
Why is that? Reads better to me like this.

Reply:
I had to read it multiple times to realize what's going on.

I don't know what reads better (because I'm not sure what exactly you're comparing it to), I just know I found this particular form to complex for my taste.

But, even for the simplest change (change "but" to a full stop"), I think it's an improvement:
```
Setting this up will highly depend on where you are deploying your Wasp project.
In general, it comes down to defining the env vars via mechanisms that your hosting provider provides.
```
I believe simpler and shorter sentences are almost always easier to read (since they give you a "checkpoint"). This is especially true for non-native speakers.